### PR TITLE
Skip empty benchmarks

### PR DESCRIPTION
**Pull request by inaki-amatria**
Adding empty benchmarks was causing the `multiple-compilers.sh` to fail with the following stack trace:

```
. . .
[100%] Built target PWR073
Traceback (most recent call last):
  File "/home/docker/open-catalog/Benchmark/run-benchmarks.py", line 215, in <module>
    main()
  File "/home/docker/open-catalog/Benchmark/run-benchmarks.py", line 206, in main
    run_output_csv(
  File "/home/docker/open-catalog/Benchmark/run-benchmarks.py", line 71, in run_output_csv
    result = json.loads(
             ^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

The problem is we would execute a benchmark binary that produced no output and `json.loads` would try to read an empty string. This would cause the exception shown above: `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`.

### CODE REVIEW

**Review by alvrogd — APPROVED** [review]
Good catch!

File: Benchmark/cmake/modules/Benchmark.cmake [hunk 1]
```diff
@@ -4,7 +4,7 @@ add_custom_target(run
 )
 
 # Utility function to automatically create the build target for the given check 
-macro(add_benchmark CHECKID)
+function(add_benchmark CHECKID)
```
Comment (daniel-otero) [inline Benchmark/cmake/modules/Benchmark.cmake:7]: Any particular reason why?

**Review by daniel-otero — APPROVED** [review]
LGTM!

File: Benchmark/cmake/modules/Benchmark.cmake (hunk 1 above)
Comment (inaki-amatria) [reply Benchmark/cmake/modules/Benchmark.cmake:7]: Or else I cannot `return()` from the function

**Pull request merged**